docfx~/animator-services.md

```diff
@@ -0,0 +1,301 @@
+if (existingLayer != null)
+{
+    existingLayer.Weight = 0.5f;
+    existingLayer.BlendingMode = AnimatorLayerBlendingMode.Additive;
+}
+```
+
+## Using AnimationIndex and ObjectPathRemapper
+
+### Path Remapping and Object Tracking
+
+The `ObjectPathRemapper` is crucial for handling object hierarchy changes during the build process:
+
+```csharp
+var pathRemapper = animatorServices.ObjectPathRemapper;
+
+// Get the virtual path for a GameObject
+string virtualPath = pathRemapper.GetVirtualPathForObject(myGameObject);
+
+// Record a new object tree (for prefab instantiation)
+GameObject newPrefab = PrefabUtility.InstantiatePrefab(somePrefab) as GameObject;
+pathRemapper.RecordObjectTree(newPrefab.transform);
+
+// Replace an object before removing it
+pathRemapper.ReplaceObject(oldObject, newObject);
+
+// Get final path mappings (typically done automatically on deactivation)
+var pathMappings = pathRemapper.GetVirtualToRealPathMap();
+```
+
+### Animation Indexing and Querying
+
+The `AnimationIndex` provides efficient ways to find and modify animations:
+
+```csharp
+var animationIndex = animatorServices.AnimationIndex;
+
+// Find all clips that affect a specific object path
+var clipsForObject = animationIndex.GetClipsForObjectPath("MyObject/ChildObject");
+
+// Find clips for a specific binding
+var binding = EditorCurveBinding.FloatCurve("MyObject", typeof(Transform), "localPosition.x");
+var clipsForBinding = animationIndex.GetClipsForBinding(binding);
+
+// Apply changes to all clips affecting an object
+animationIndex.ProcessClipsForObjectPath("MyObject", (clip, binding) =>
+{
+    // Modify the clip/binding as needed
+    if (binding.propertyName == "localPosition.x")
+    {
+        var curve = clip.GetFloatCurve(binding);
+        // Modify curve...
+        clip.SetFloatCurve(binding, curve);
+    }
+});
+```
+
+### Advanced Path Remapping
+
+```csharp
+// Create custom path remapping rules
+var pathMappings = new Dictionary<string, string?>
+{
+    ["OldPath/Object"] = "NewPath/Object",  // Rename path
+    ["DeletedPath/Object"] = null,          // Delete path (remove from animations)
+    ["AnotherPath"] = "ReplacementPath"     // Replace path
+};
+
+// Apply the remapping (this happens automatically on deactivation)
+animationIndex.RewritePaths(pathMappings);
+```
+
+## Complete Example: Adding a Toggle Animation
+
+Here's a complete example that demonstrates adding a toggle animation to a VRChat avatar:
+
+```csharp
+[DependsOnContext(typeof(AnimatorServicesContext))]
+public class AddToggleAnimationPass : Pass<AddToggleAnimationPass>
+{
+    protected override void Execute(BuildContext context)
+    {
+        var animatorServices = context.Extension<AnimatorServicesContext>();
+        var controllerContext = animatorServices.ControllerContext;
+        var pathRemapper = animatorServices.ObjectPathRemapper;
+        
+        #if NDMF_VRCSDK3_AVATARS
+        // Get the FX layer
```

Comment:
Add a note to this comment: The animator services context will automatically create a default FX layer, so this should not normally happen when performing a build for a VRChat avatar.